src/compiler/crystal/macros/methods.cr

```diff
@@ -927,6 +927,16 @@ module Crystal
         interpret_argless_method(method, args) { @of.try(&.value) || Nop.new }
       when "type"
         interpret_argless_method(method, args) { @name || Nop.new }
+      when "delete"
+        interpret_one_arg_method(method, args) do |key|
+          unless key.is_a? Crystal::NumberLiteral | Crystal::SymbolLiteral | Crystal::StringLiteral | Crystal::BoolLiteral
+            key.raise "argument to HashLiteral#delete must be a number, bool, symbol, or string, not #{key.class_desc}:\n\n#{key}"
```

Comment:
Why?

Reply:
My thinking was these are really the only things you can have as keys in macro level hash literals.  So it would error in the case if you mistakenly pass `nil` or like another hash or array or something.

Reply:
You can have any ast node as a key, why not?

Reply:
Fair point.  I'll push that up later today.

Reply:
Thanks!